### PR TITLE
Add resource sync changelist

### LINKED ISSUE #61: ResourceSync implementation should support change lists
Decomposed from projecthydra/sufia#2915.

Add support for change lists: http://www.openarchives.org/rs/1.0.9/resourcesync#DesChanges

### PR DESCRIPTION
**Pull request by jcoyne**
Fixes #61

### CODE REVIEW

File: app/search_builders/hyrax/abstract_type_relation.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    def equivalent_class?(klass)
+      allowable_types.include?(klass)
+    end
+
+    def search_model_clause
+      clauses = allowable_types.map do |k|
+        ActiveFedora::SolrQueryBuilder.construct_query_for_rel(has_model: k.to_s)
+      end
+      clauses.size == 1 ? clauses.first : "(#{clauses.join(' OR ')})"
```
Comment (dbrower) [inline app/search_builders/hyrax/abstract_type_relation.rb]: Doesn't `clauses.join(' OR ')` do the right thing when there is only one item in the list?

File: app/search_builders/hyrax/exposed_models_relation.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+module Hyrax
+  # A relation that scopes to all user visible models (e.g. works + collections + file sets)
+  class ExposedModelsRelation < AbstractTypeRelation
+    def allowable_types
+      Hyrax.config.curation_concerns + [Collection, ::FileSet]
```
Comment (mjgiarlo) [inline app/search_builders/hyrax/exposed_models_relation.rb]: Why collection and not admin set?

File: app/search_builders/hyrax/abstract_type_relation.rb (hunk 1 above)
Comment (mjgiarlo) [inline app/search_builders/hyrax/abstract_type_relation.rb]: Yes, but the false expression in the ternop also builds a string with parentheses.

Comment (dbrower) [inline app/search_builders/hyrax/abstract_type_relation.rb]: And it is an error to include parentheses if there is not an `OR`...? this is only a nitpick.

Comment (jcoyne) [inline app/search_builders/hyrax/abstract_type_relation.rb]: @dbrower this isn't logic I'm changing. I've just moved it from a concrete class to an abstract class.

File: app/search_builders/hyrax/exposed_models_relation.rb (hunk 2 above)
Comment (jcoyne) [inline app/search_builders/hyrax/exposed_models_relation.rb]: We've never exposed AdminSets via resource synch before, but we have Collections. I'm trying to be consistent with what is already there.

File: app/search_builders/hyrax/abstract_type_relation.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    end
+
+    class DummyModel
```
Comment (jeremyf) [inline app/search_builders/hyrax/abstract_type_relation.rb]: Is this class something that we won't be exposing to the public? If so, what are your thoughts on adding a `private_constant :DummyModel` after the close of the class?

File: lib/hyrax/resource_sync/change_list_writer.rb [hunk 4]
```diff
@@ -0,0 +1,90 @@
+module Hyrax
+  module ResourceSync
+    # TODO: the big assumption I'm making here is that the repository has fewer
+    # than 50,000 resources to list. The Sitemap protocol is limited at 50,000
+    # items, so if we require more than that, we must have multiple Change
+    # lists and add a Change List Index to point to all of them.
+    class ChangeListWriter
```
Comment (jeremyf) [inline lib/hyrax/resource_sync/change_list_writer.rb]: Can we use pagination for generating sitemap generation?

I was crawling this site http://grognardia.blogspot.com/sitemap.xml when I discovered the sitemap leveraged pagination.

Comment (jcoyne) [inline lib/hyrax/resource_sync/change_list_writer.rb]: We absolutely need too, but each page can have 50k objects. See https://github.com/projecthydra-labs/hyrax/issues/59 if you'd like to work on that.

File: app/search_builders/hyrax/abstract_type_relation.rb (hunk 3 above)
Comment (jcoyne) [inline app/search_builders/hyrax/abstract_type_relation.rb]: This PR was already merged, but feel free to add that.

**Pull request merged**